pkg/tsdb/prometheus/models/query.go

```diff
@@ -90,16 +80,18 @@ func Parse(query backend.DataQuery, timeInterval string, intervalCalculator inte
 	// Interpolate variables in expr
 	timeRange := query.TimeRange.To.Sub(query.TimeRange.From)
 	expr := interpolateVariables(model, interval, timeRange, intervalCalculator, timeInterval)
-	rangeQuery := model.RangeQuery
-	if !model.InstantQuery && !model.RangeQuery {
+	rangeQuery := model.Range
+	if rangeQuery != nil && model.Instant != nil && model.
+		Range != nil && !*model.
+		Instant && !*model.Range {
```

Comment:
nil checks are not nice but for now if we have optional parameters we have to have pointers. we have to get rid of some optional but eventually we will have some. There must be a better way to deal with nil checks.

Reply:
in general, CUE does give more nuanced control over values, optionality and defaults than Go. Once we put its unmarshaler in place here, there will be less ambiguity in the values that appear in your Go structs.